components/motor/gpio/loop.go

```diff
@@ -2,44 +2,59 @@ package gpio
 
 import (
 	"context"
+	"math"
+	"sync"
+	"time"
 
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"go.viam.com/rdk/components/encoder"
+	"go.viam.com/rdk/components/motor"
 	"go.viam.com/rdk/control"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
 )
 
 // SetState sets the state of the motor for the built-in control loop.
-func (m *EncodedMotor) SetState(ctx context.Context, state []*control.Signal) error {
-	if m.loop != nil && !m.loop.Running() {
+func (cm *controlledMotor) SetState(ctx context.Context, state []*control.Signal) error {
+	if cm.loop != nil && !cm.loop.Running() {
 		return nil
 	}
 	power := state[0].GetSignalValueAt(0)
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	return m.setPower(ctx, power)
+	cm.mu.Lock()
+	defer cm.mu.Unlock()
+	return cm.real.SetPower(ctx, power, nil)
 }
 
 // State gets the state of the motor for the built-in control loop.
-func (m *EncodedMotor) State(ctx context.Context) ([]float64, error) {
-	pos, err := m.position(ctx, nil)
+func (cm *controlledMotor) State(ctx context.Context) ([]float64, error) {
+	ticks, _, err := cm.enc.Position(ctx, encoder.PositionTypeTicks, nil)
```

Comment:
[supernit] at some point we should validate that the position type returned matches the position type we requested. 

Since the position type is part of the response, it seems reasonable for a user to implement an encoder that doesn't error if your type is not supported

Reply:
I see what you mean. we do already check that the encoder supports ticks, but you mean if the encoder always returns absolute or something and ignores whatever the requested type is?

Reply:
yeah pretty much that, though I missed that we made a properties request as part of the contructor